### PR TITLE
adding multi series aggregation method

### PR DESCRIPTION
**Pull request by mrgaaron**
Forgot to put this method in the new client.  Correcting that oversight.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     author_email="dev@tempo-db.com",
     url="http://github.com/getsentry/tempodb-python/",
     description="A client for the TempoDB API",
-    packages=["tempodb"],
+    packages=["tempodb", "tempodb.temporal", "tempodb.validate"],
```
Comment (myagley) [inline setup.py]: Why was this line needed?

File: tempodb/client.py [hunk 2]
```diff
@@ -242,6 +242,53 @@ def read_data(self, key, start=None, end=None, fold=None,
                                      tz=data['tz'])
         return c
 
+    def aggregate_data(self, aggregation, keys=[], tags=[], attrs={},
```
Comment (myagley) [inline tempodb/client.py]: We should support an optional rollup on this call as well.

File: setup.py (hunk 1 above)
Comment (mrgaaron) [inline setup.py]: The packages declaration isn't recursive.  Without that, the temporal and
validate packages fail to go into the library installed in site-packages
and you end up with import errors.

2014/1/15 Mike Yagley notifications@github.com

Comment (meshulam) [inline setup.py]: Pretty sure this should actually be: 
packages=["tempodb", "tempodb.temporal", "tempodb.protocol"],

setup.py failed for me otherwise.

File: tempodb/client.py [hunk 3]
```diff
@@ -242,6 +247,103 @@ def read_data(self, key, start=None, end=None, fold=None,
...
+        url_args = endpoint.make_url_args(params)
+        url = '?'.join([url, url_args])
+        resp = self.session.get(url)
+        r = Response(resp, self.session)
+        data = json.loads(r.resp.text)
+        c = protocol.DataPointCursor(data['data'], protocol.DataPointFound, r,
+                                     tz=data['tz'])
+        return c
+
+    def aggregate_data(self, aggregation, keys=[], tags=[], attrs={},
```
Comment (myagley) [inline tempodb/client.py]: We should allow an optional `rollup` on this call as well.

Comment (mrgaaron) [inline tempodb/client.py]: Does that need to be rollup.period and rollup.function as in the other calls, or is it just rollup for this one?

Comment (myagley) [inline tempodb/client.py]: Yeah it should look like the single series call, which I believe takes both a period and folding function. You can kind of think of it as a pipeline where the aggregation creates a single series that can then be rolled up.

File: tempodb/client.py [hunk 4]
```diff
@@ -86,8 +88,10 @@ class Client(object):
     :param string key: your API key
     :param string secret: your API secret"""
 
-    def __init__(self, key, secret, base_url=endpoint.BASE_URL):
-        self.session = endpoint.HTTPEndpoint(key, secret, base_url)
+    def __init__(self, database_id, key, secret, base_url=endpoint.BASE_URL):
```
Comment (meshulam) [inline tempodb/client.py:126]: Should database_id be optional to maintain compatibility?

Comment (myagley) [inline tempodb/client.py:126]: No it should be required. This v1.0 of the client has no backwards compatibility requirement. We are future proofing this client for multiple credentials per database. In order to support that, we need to be able to separate referencing a database from the api key. We probably should update the website to display database_id, api_key and api_secret.

File: tempodb/client.py (hunk 3 above)
Comment (mrgaaron) [inline tempodb/client.py]: Generally speaking, we aren't worried about backwards compatibility with these clients.  On the other hand, this change would probably impact adoption if we were to release the client before the database ID auth feature goes in.  So that's more a question of release strategy, but as we get along with these client rewrites it looks more and more like we're going to be waiting for the official 1.0 rollout before getting them out there.

File: tempodb/protocol/objects.py [hunk 5]
```diff
@@ -272,3 +272,44 @@ def to_dictionary(self):
...
+        :rtype: None"""
+
+        if type(json_text) in [str, unicode]:
+            j = json.loads(json_text)
+        else:
+            j = json_text
+
+        try:
+            for p in self.properties:
+                print p
```
Comment (meshulam) [inline tempodb/protocol/objects.py]: Remove print statement

File: tempodb/client.py (hunk 3 above)
Comment (meshulam) [inline tempodb/client.py]: Ok, makes sense. Then the pydoc should be updated to add a description of
the database_id param.

## 

Matt Meshulam
Solutions Architect, TempoDB
https://tempo-db.com

Comment (myagley) [inline tempodb/client.py]: I agree. We need to update the pydoc as well as the manage page on the website to list the database_id. Initially all database_id and api_key's will be the same (for each database). This will diverge after the multiple credentials feature is released. But, if we require the param from the start, no one will need to update their code when that feature is rolled out.

We are concerned with backwards compatibility for minor and patch releases, but not for major releases. This is a major release. 0.x->1.0
